### DISCUSSION TITLE
Fonts in Android app

### DISCUSSION THREAD

**Discussion by BorislavLuis** [ocornut/imgui#6418] [New Users Build/Link/Run issues ONLY!]
Hello. How can I use greek and cyrillic letters in an android app? I think that all of my setting are fine. All the settings in the Android Studio project editor are set to UTF-8, in my Manifest UTF-8 is set , and also in my build.grandle UTF-8 is set. In the cmake file utf-8 is also set but when the app starts all of the text is with question marks. Is it possible to use greek and cyrillic letters in android app with native code?

**Comment by ocornut** [comment]
This can come from two things,

1. Call `ImGui::DebugTextEncoding("....");` with your string to confirm if your UTF-8 encoding is correct. If not it's an encoding problem, if correct it's a font loading/ranges problem.
2. Use Demo->Tools->Metrics->Fonts to confirm that your font was loaded with the aforementioned characters ranges.

(Neither are specific to Android)

**Reply by BorislavLuis** [reply]
Thank you for the fast response. It wasn't an encoding problem. The problem was from the glyph. I added new version of ImGui with GetGlyphRangesGreek() and everything is fine. Cyrillic and greek letters appear good.

**Reply by ocornut** [reply]
Today I have reworded this document:
https://github.com/ocornut/imgui/blob/master/docs/FONTS.md
To put more emphasis on common problems.

I hope to get rid of those pre-loaded glyphes at some point!
